Florian

Florian-collab opened the pull request:
send code changes to the project

YannRennes2 committed:
Update CGH.py

print test

ArthurCns committed:
Update CGH.py

YannRennes2 committed:
Revert "Update CGH.py"

This reverts commit 8723a892a6aa86acaf6ee45356d640537760e59b.

ArthurCns committed:
Merge branch 'master' of https://github.com/Florian-collab/graphh

ArthurCns committed:
Revert "Update CGH.py"

This reverts commit 63bc76d1e47238c0ec8051bf777560fd57b5f689.

YannRennes2 committed:
Erreur de clé d'adresse et warnings

Mise à jour

Florian-collab committed:
Create mygraphhv2

Florian-collab committed:
Add files via upload

Florian-collab committed:
Delete mygraphhv2

Florian-collab committed:
Delete testaarthur.py

Florian-collab committed:
Update mygraphhv2.py

Florian-collab committed:
Update mygraphhv2.py

Florian-collab committed:
Update myCGHError.py

Florian-collab committed:
Update CGH.py

Florian-collab committed:
Update CGHError.py

Florian-collab commented:
I change CGH and CGHerror

rtavenar commented on myCGHError.py:
```diff
@@ -0,0 +1,87 @@
+import requests
```
Hum, I guess this file should not be included here. Please remove it.

rtavenar commented on mygraphhv2.py:
```diff
@@ -0,0 +1,531 @@
+import requests
```
Here, also, this is a duplicate of `CHG.py`, please include only one of them.

rtavenar commented on graphh/CGH.py:
```diff
@@ -1,45 +1,48 @@
+class GraphHopper:
+    """GraphHopper API class.
+
+        Parameters
```
Indentation level for the docstring should be that of the beginning of the string (same level as the initial `"""`), cf old version if needed.

rtavenar commented on graphh/CGH.py:
```diff
@@ -1,45 +1,48 @@
+            Whether the account corresponding to this key is a premium account
+            or not
+    """
+    """"
```
This line is useless (`""""`)

rtavenar commented on graphh/CGH.py:
```diff
@@ -1,45 +1,48 @@
+         premium: bool
+            Whether the account corresponding to this key is a premium account
+            or not
+    """
```
This should not be the end of the docstring (examples are just below), so please do not close it with the `"""`.

rtavenar commented on graphh/CGH.py:
```diff
@@ -48,42 +51,84 @@ class GraphHopper(object):
+
+
+    def _url_requests(self, api, data, request="get"):
+
```
There should be no blank line between the function definition and the docstring (probably necessary for automatic doc generation to work properly).

rtavenar commented on graphh/CGH.py:
```diff
@@ -48,42 +51,84 @@ class GraphHopper(object):
-            list of parameters to insert in the url
+        data: dict
+            dict of parameters to insert in the url
 
```
Please add the `request` argument to the docstring and document it properly.

rtavenar replied:
There is no `s` at the end of `request`: be careful when you name arguments: what you write here is used to generate the docs.

rtavenar commented on graphh/CGH.py:
```diff
@@ -48,42 +51,84 @@ class GraphHopper(object):
-        complete_url += "&key=" + self.api_key
+
+        if request.upper() == "POST":
+            complete_url += "key={}".format(self.api_key)
```
Is there a reson why the key is not added to the data dict as for the GET query?

rtavenar commented on graphh/CGH.py:
```diff
@@ -48,42 +51,84 @@ class GraphHopper(object):
-            CGHError.CGHError(e)
+            reponse.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+             myCGHError.CGHError(e)
```
Please put all your necessary code in `CGHError` and call only functions from this module.

rtavenar replied:
Here you call `myCGHError.CGHError`: check all occurrences of `myCGHError` in your code, they should all be removed.

rtavenar commented on graphh/CGH.py:
```diff
@@ -183,94 +235,138 @@ def route(self, l_latlong , vehicle="car", locale="en",
+
+
+
+    def maxtrix(self,l_from_address, l_to_address,out_array, format="default", vehicle="car", request="get"):
```
Hum, the name for this method is probably not the one you wanted to use...

rtavenar commented on graphh/CGH.py:
```diff
@@ -183,94 +235,138 @@ def route(self, l_latlong , vehicle="car", locale="en",
-            The unit of the distance returned chosen between "ms", "s", "min"
-            and "h"
-            By default the unit will be in milliseconds
 
```
Please document the `format` parameter :)

rtavenar reviewed (commented):
This is really going in the right direction. My main concern is that you should store all the necessary code in `CGH.py` and `CGHError.py` and remove the other 2 files.

And then I have some formatting concerns that we can deal with step by step through the review process.

Good job, keep going on!

Florian-collab committed:
Update CGH.py

Just edit the docstring errors

Florian-collab committed:
Update CGH.py

Added precisions to the warning and the matrix, corrected english and syntax mistakes.

Florian-collab committed:
Update CGHError.py

Added correction of matrix (dimension).

Florian-collab committed:
Delete myCGHError.py

Florian-collab committed:
Delete mygraphhv2.py

rtavenar commented:
Could you please take all my comments from my previous review into account and upload an adapted version? 

Also, as you write it, you code adds a new dependency to the package (`pandas`). It would be better to have something like:

```python
try:
  import pandas
  PANDAS_IMPORTED = True
except ImportError:
  PANDAS_IMPORTED = False

[...]

def function_in_which_I_need_pandas(...):
  if not PANDAS_IMPORTED:
    raise ValueError("pandas was not imported correctly, hence function_in_which_I_need_pandas is not available")
```

Florian-collab committed:
Update CGH.py

We added a dependency  concept between pandas et numpy

Florian-collab committed:
Update CGHError.py
